packages/truffle-debugger/lib/data/selectors/index.js

```diff
@@ -235,13 +235,23 @@ const data = createSelectorTree({
       refs: createLeaf(
         [
           "/proc/assignments",
-          "./_"
+          "./_",
+          solidity.current.functionDepth //for pruning things too deep on stack
         ],
 
-        (assignments, identifiers) => Object.assign({},
+        (assignments, identifiers, curDepth) => Object.assign({},
           ...Object.entries(identifiers)
             .map( ([identifier, id]) => {
-              let { ref } = (assignments[id] || {})
+              let matchIds = Object.keys(assignments).filter(
+                  (longId) => longId.endsWith(":"+id) //get cases of that var
+                  ).map(
+                  (longId) => longId.split(":")[0] //get just the stack frame
+                  );
+              //want innermost but not beyond current depth
+              let maxMatch=Math.min(curDepth,Math.max(...matchIds));
+                  //note: if no matches, will return -Infinity
+                  //however the return value in this case is irrelevant
```

Comment:
Style suggestion: 

Typically for comments that go on their own line, I'd say put them like this:

```javascript
    // above line or block of code with same indendation level
    // note: side node
    let maxMatch = Math.min(...);
    const otherLine = alsoRelatedToComment;

    // another block of code with a comment
    return { ... }
```

Note empty-line spacing to visually separate commented blocks of code.

Oh yeah, also whitespace rules:
- Whitespace on both sides of binary operators
- Whitespace after `,` in argument lists
- Whitespace at start, end, and in-between items, for array and object literals

Reply:
Heh -- a number of my whitespace omissions were to get things (with their comments) to fit on one line.  A lot of comments are going to have to become separate lines now, I guess...

Reply:
Yeah, I play that game often, trying to get comments to fit on the same line, etc.

I like lots of short lines with empty lines in between logical sections.

When I use inline comments, I try to make sure it's just a tiny note.

Reply:
OK, done, except that I have some comments which go on their own line but don't seem substantial enough to warrant whitespace before, so I didn't put whitespace before those.